### DISCUSSION TITLE
Combine affine transformations

### DISCUSSION THREAD

**Discussion by Spenhouet** [Project-MONAI/MONAI#2427] [Ideas]
**Is your feature request related to a problem? Please describe.**

All affine transformations perform an interpolation. In medical imaging this is very crucial since you loose information with every interpolation.
Since most libraries take this into account and collect and combine affine transformations to only perform a single transformation at one timepoint, we assumed that Compose is exactly doing that.
We were quite shocked to discover that MONAI actually does not do this and instead performs an interpolation on every individual transformation step (were an interpolation is required).

MONAI comes with a lot of functionality that we need and is a great tool. The fact above really shocked us since we assumed that this would be taken into account since MONAI is specifically for medical image data.
Was this overlooked during development? How come that this could happen?

**Describe the solution you'd like**
All spatial transformations should be collected and then combined down to as few transformations as possible. Especially for affine transformations.

**Comment by wyli** [comment]
thanks, great to see this type of dicussions, yes it is important and has been discussed since day one https://github.com/Project-MONAI/MONAI/issues/112

Currently MONAI provides specific transforms to do the fused spatial transformations:
https://github.com/Project-MONAI/MONAI/blob/a0afa604bbb5f4e673597c5ca833ba1fd4b8547e/monai/transforms/spatial/dictionary.py#L555-L573
https://github.com/Project-MONAI/MONAI/blob/a0afa604bbb5f4e673597c5ca833ba1fd4b8547e/monai/transforms/spatial/dictionary.py#L826-L846

This is a generic discussion, I'm converting it to the discussion tab.

**Reply by Spenhouet** [reply]
@wyli Good to hear that this is already on the radar.

We are already using the dictionary based transforms in our pipeline. Are you saying that the `Affined` transform already performs this fusing? It did not look like it to me but I might have misunderstood the code.

**Reply by Nic-Ma** [reply]
Hi @Spenhouet ,

Glad to see your interest and experiments with MONAI.
For more details of the fused transform(Affine, Elastic deformation, etc.), you can take a look at:
https://github.com/Project-MONAI/tutorials/blob/master/modules/3d_image_transforms.ipynb
https://github.com/Project-MONAI/tutorials/blob/master/acceleration/transform_speed.ipynb

Thanks.
